### ISSUE TITLE
Add 3D output (anaglyph and SBS) when using BAM headtracking

### ISSUE THREAD

**Issue by PG248** [vpinball/vpinball#1259]
**Is your feature request related to a problem? Please describe.**
I first thought it is a bug but was told it never existed in Visual Pinball. 3D output neither anaglyph nor SBS works together with BAM headtracking.

Both work together in Future Pinball but you have to choose between headtracking and 3D in VPX. As soon as headtracking is switched on only the picture for the left eye is displayed in 3D mode. 

**Describe the solution you'd like**
If you ever tried 3D and headtracking together (in Future Pinball) you know that only this combination gives you the ultimate realism of playing a real pin. It is as immersive as playing in VR but without being locked out from the real world. You can play with multiple people and all see the table in 3D (ok only the player benefits from the head tracking).
Maybe implementing it is harder than I can imagine but as 3D fully works without headtracking and headtracking works in 3D but only for the left eye, it feels like all necessary components are already there but some connection between them is missing.
If some developer who loves 3D and headtracking as much as I do could find the loose end and connect them, the ultimate reality experience in Visual Pinball would become true. 

**Describe alternatives you've considered**
I was told fake stereo may work with head tracking. I have to test this. But even if that is true, real 3D with headtracking would be preferable.

**Comment by gjupp** [comment]
Yeah, hopefully this gets some attention at some point in the future as it would be very nice to have. In the meantime, I suggest trying BAM with Reshade/Depth3D and let that shader handle the 3D as it's better than the inbuilt fake 3D mode, but doesn't support the transparent surfaces depth correctly compared to the true stereo 3D mode of GL VPX version. Make sure that the 3D setting is disabled in VPX graphics settings.

I think this combo will likely work but I haven't tested it myself.

**Comment by PG248** [comment]
Thank you for reminding me of BAM with Reshade/Depth3D. I'll try that as well in addition to fake 3D mode. To use Reshade I have to run the DIrectX version of VPX, right?

**Comment by gjupp** [comment]
> Thank you for reminding me of BAM with Reshade/Depth3D. I'll try that as well in addition to fake 3D mode. To use Reshade I have to run the DIrectX version of VPX, right?

Ive only ever run reshade with DirectX version, but it will probably work with GL version as well, you would just need to install reshade with that API option.  It may run faster with GL, not sure. Depth3D is certainly better than inbuilt fake 3D mode though. Be sure to disable any inbuilt 3D mode when running Depth3D.

**Comment by PG248** [comment]
Tried Reshade/Depth3D together with BAM headtracking with VPX 10.8 GL 64 bit. It works but there was noticeable stuttering when moving my head. So far I get smooth performance with anaglyph 3D and BAM headtracking only with Future Pinball.

**Comment by gjupp** [comment]
> Tried Reshade/Depth3D together with BAM headtracking with VPX 10.8 GL 64 bit. It works but there was noticeable stuttering when moving my head. So far I get smooth performance with anaglyph 3D and BAM headtracking only with Future Pinball.

Make sure the performance mode box is ticked in Depth3D shader if not already.

**Comment by toxieainc** [comment]
@vbousquet How difficult would this be to add for 10.8? Sounds reasonably simple, but you know the pipeline best nowadays.

**Comment by vbousquet** [comment]
I can see 2 ways of quickly implementing this:
- change BAM camera setup to support anaglyph (somewhat complex, and will likely be somewhat dropped if we move to the plugin arch for this type of thing),
- change BAM camera setup to use VPX camera setup (remove most of BAM config which is now duplicated inside VPX, just update the player position based o BAM acquisition).

Since I don't have a BAM setup and plenty of other things going on, this is not something I will do but if someone is willing too, I'm sure lots of people will be happy!

**Comment by PG248** [comment]
@vbousquet now that you have a kinect 2 I hope you may take a second look at making headtracking work with 3D. Personally I think your second suggestion to just update the player position based on BAM acquisition is the easier solution and should work equally well with both anaglyph and 3D interlaced. 
Headtracking really helps to prevent warping in 3D and both together allow a realism otherwise only possible with VR.

**Comment by 0ldG1t00** [comment]
Can anyone confirm if BAM and 3D both work together and if so have gotten good results and how was it setup.
